java/test/jmri/jmrit/display/layoutEditor/LayoutEditorChecksTest.java

```diff
@@ -1,40 +1,349 @@
+    private String checkUnConnectedTracksMenuTitle = Bundle.getMessage("CheckUnConnectedTracksMenuTitle");
+    private String checkUnBlockedTracksMenuTitle = Bundle.getMessage("CheckUnBlockedTracksMenuTitle");
+    private String checkNonContiguousBlocksMenuTitle = Bundle.getMessage("CheckNonContiguousBlocksMenuTitle");
+    private String checkUnnecessaryAnchorsMenuTitle = Bundle.getMessage("CheckUnnecessaryAnchorsMenuTitle");
+
+    //name strings
+    private static String myBlockName = "My Block";
+    private static String rightHandName = "Right Hand";
+    private static String leftHandName = "Left Hand";
+
+    //jemmy operators
+    private static EditorFrameOperator layoutEditorEFO = null;
+
+    private static LayoutEditor layoutEditor = null;
+    private static LayoutEditorChecks layoutEditorChecks = null;
+    private static LayoutBlock layoutBlock = null;
+    private static LayoutTurnout ltRH = null, ltLH = null;
+    private static PositionablePoint a1 = null;
+    private static PositionablePoint a2 = null;
+    private static TrackSegment ts1 = null;
+    private static TrackSegment ts2 = null;
+    private static TrackSegment ts3 = null;
+
     @Test
     public void testCTor() {
         Assume.assumeFalse(GraphicsEnvironment.isHeadless());
-        LayoutEditor le = new LayoutEditor("Layout Editor Checks Test Layout");
-        LayoutEditorChecks t = new LayoutEditorChecks(le);
-        Assert.assertNotNull("exists", t);
-        JUnitUtil.dispose(le);
+        Assert.assertNotNull("exists", layoutEditorChecks);
     }
 
-    // The minimal setup for log4J
-    @Before
-    public void setUp() {
+    @Test
+    public void testSetup() {
+        Assume.assumeFalse(GraphicsEnvironment.isHeadless());
+        Assert.assertNotNull("layoutEditor not null", layoutEditor);
+
+        //everything not null?
+        Assert.assertNotNull("layoutBlock not null", layoutBlock);
+        Assert.assertNotNull("ltRH not null", ltRH);
+        Assert.assertNotNull("ltLH not null", ltLH);
+        Assert.assertNotNull("a1 not null", a1);
+        Assert.assertNotNull("a2 not null", a2);
+        Assert.assertNotNull("ts1 not null", ts1);
+        Assert.assertNotNull("ts2 not null", ts2);
+        Assert.assertNotNull("ts3 not null", ts3);
+
+        //everything named correctly?
+        Assert.assertEquals("layoutBlock.getUserName()", myBlockName, layoutBlock.getUserName());
+        Assert.assertEquals("ltRH.getName()", rightHandName, ltRH.getName());
+        Assert.assertEquals("ltLH.getName()", leftHandName, ltLH.getName());
+        Assert.assertEquals("a1.getName()", "A1", a1.getName());
+        Assert.assertEquals("a2.getName()", "A2", a2.getName());
+        Assert.assertEquals("ts1.getName()", "T1", ts1.getName());
+        Assert.assertEquals("ts2.getName()", "T2", ts2.getName());
+        Assert.assertEquals("ts3.getName()", "T3", ts3.getName());
+    }
+
+    @Test
+    public void testBundleStrings() {
```

Comment:
Testing package Specific bundle strings really should be in the bundle tests, not here.

Reply:
Personally, I'd prefer to test the ones used in the test class that uses them… but I'm not married to it so… removed.